arch/arm/core/cortex_m/vector_table.S

```diff
@@ -69,3 +69,80 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 #else
     .word __reserved
 #endif
+
+#if !defined(CONFIG_CPU_CORTEX_M_HAS_VTOR)
```

Comment:
I think here need change to
#if ( !defined(CONFIG_CPU_CORTEX_M_HAS_VTOR) && defined(CONFIG_IS_BOOTLOADER) )